packages/fetch/src/types.ts

```diff
@@ -16,182 +16,174 @@
-   */
-  new (): AbortController;
+  readonly clone: () => this;
 
-  /**
-   * Returns the AbortSignal object associated with this object.
-   */
-  readonly signal: AbortSignal;
-  /**
-   * Invoking this method will set this object's AbortSignal's aborted flag and signal to any observers that the associated activity is to be aborted.
-   */
-  abort(): void;
-};
+  /*
+  static error(): Response;
+  static json(data: any, init?: ResponseInit): Response;
+  static redirect(url: string | URL, status: ResponseRedirectStatus): Response;
+  */
+}
 
 /**
  * Minimal types for a ReadableStream.
  * @todo Missing declarations for `constructor`, `pipeThrough`, `pipeTo` and `tee`.
  */
 
-export type ReadableStream = {
+export type ReadableStream<T = unknown> = {
   /**
    * Creates a reader and locks the stream to it.
    * While the stream is locked, no other reader can be acquired until this one is released.
    * @see https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream/getReader
    */
-  getReader(): StreamReader;
+  getReader(): ReadableStreamDefaultReader<T>;
+  getReader(options: { mode: "byob" }): unknown;
 
   /**
    * Cancel is used when you've completely finished with the stream and don't need any more data from it, even if there are chunks enqueued waiting to be read.
    * @return a Promise that resolves when the stream is canceled.
    * @see https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream/cancel
    */
   cancel: (reason?: string) => Promise<void>;
-} & AsyncIterable<Uint8Array>;
+}; // & AsyncIterable<T>;
+
+export type ReadableStreamReadDoneResult = {
+  done: true;
+  // TODO: Make this `value: undefined` when https://github.com/microsoft/TypeScript-DOM-lib-generator/pull/1676 is merged and released
+  value?: unknown;
+};
 
-export type StreamReader = {
+export type ReadableStreamReadValueResult<T> = {
+  done: false;
+  value: T;
+};
+
+export type ReadableStreamReadResult<TValue = unknown> =
+  | ReadableStreamReadValueResult<TValue>
+  | ReadableStreamReadDoneResult;
+
+export type ReadableStreamDefaultReader<T = unknown> = {
   /**
    * a Promise that fulfills when the stream closes, or rejects if the stream throws an error or the reader's lock is released.
    */
-  closed: Promise<boolean>;
+  closed: Promise<void>;
   /**
    * Cancel is used when you've completely finished with the stream and don't need any more data from it, even if there are chunks enqueued waiting to be read.
    * @param reason A human-readable reason for the cancellation.
    * @returns a Promise that resolves when the stream is canceled. Calling this method signals a loss of interest in the stream by a consumer.
    * @see https://developer.mozilla.org/en-US/docs/Web/API/ReadableStreamDefaultReader/cancel
    */
-  cancel(reason?: string): Promise<string | undefined>;
+  cancel(reason?: string): Promise<unknown>; // TODO: Should have returned Promise<string | undefined, but node types conflict
```

Comment:
Is there an open issue for this?

Reply:
Actually ... it seems [MDN](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream/cancel#return_value) is not following [the specification](https://streams.spec.whatwg.org/#readable-stream-cancel).

I'll have to report an issue with them, but for now I've just made this `Promise<void>`.